diff --git a/src/slvs/CMakeLists.txt b/src/slvs/CMakeLists.txt
index d931f8f..850a8a5 100644
--- a/src/slvs/CMakeLists.txt
+++ b/src/slvs/CMakeLists.txt
@@ -50,13 +50,16 @@ if(EMSCRIPTEN)
     target_link_options(slvs-wasm PRIVATE
         "SHELL:-s MODULARIZE=1"
         "SHELL:-s EXPORT_NAME=solvespace"
-        "SHELL:-s SINGLE_FILE=1"
-        "SHELL:-s INITIAL_MEMORY=512MB"
+        "SHELL:-s SINGLE_FILE=0"
+        "SHELL:-s EXPORT_ES6=1"
+        "SHELL:-s EXPORTED_RUNTIME_METHODS=HEAPU8"
+        "SHELL:-s ENVIRONMENT=web,worker,node"
+        "SHELL:-s INITIAL_MEMORY=32MB"
         "SHELL:-s ALLOW_MEMORY_GROWTH"
         -O3
         -flto
-        --closure 1)
-    set_target_properties(slvs-wasm PROPERTIES OUTPUT_NAME "slvs" SUFFIX ".js")
+        --closure 0)
+    set_target_properties(slvs-wasm PROPERTIES OUTPUT_NAME "slvs" SUFFIX ".mjs")
 endif()
 
 add_library(slvs SHARED)
